### PR TITLE
Refactor scopes in promotions and write specs for same

### PR DESCRIPTION
**Pull request by nishant-cyro**

### CODE REVIEW

**Comment by priyank-gupta** [comment]
:+1:

File: core/spec/models/spree/promotion_spec.rb [hunk 1]
```diff
@@ -26,34 +26,55 @@
...
-  end
 
-  describe ".applied" do
-    it "scopes promotions that have been applied to an order only" do
-      promotion = Spree::Promotion.create! name: "test", code: ''
-      expect(Spree::Promotion.applied).to be_empty
+    describe '.applied' do
+      let!(:promotion_not_applied) { Spree::Promotion.create! name: 'test', code: '' }
+      let(:order) { create(:order) }
+      let!(:promotion_applied) { Spree::Promotion.create! name: 'test1', code: '' }
```
Comment (Mafi88) [inline core/spec/models/spree/promotion_spec.rb]: I think that if it's named ```promotion_applied```, it should be already returned as applied. Or do you disagree?

Comment (nishant-cyro) [inline core/spec/models/spree/promotion_spec.rb]: @Mafi88 - done!

**Comment by priyank-gupta** [comment]
@nishant-cyro I think these changes are also addressed in https://github.com/spree/spree/pull/7031/files (spec missing). Lets merge that first and then you can rebase from master to submit specs from this PR.

**Pull request merged**